d2core/d2ui/textbox.go

```diff
@@ -77,7 +86,21 @@ func (v *TextBox) OnKeyChars(event d2interface.KeyCharsEvent) bool {
 	newText := string(event.Chars())
 
 	if len(newText) > 0 {
-		v.text += newText
+		if v.isNumberOnly {
+			number, err := strconv.Atoi(v.text + newText)
+			if err != nil {
+				v.Debugf("Unable to convert string %s to intager: %s", v.text+newText, err)
+				return false
+			}
+
+			if number <= v.maxValue {
```

Comment:
shouldn't this be `number < v.maxValue` ?

Reply:
not, because, it is something like that: if we have e.g 25 gold, then if we type 25 it is ok, or?

Reply:
absolutely! i did not read the code enough, sorry.